cmd/thermal-recorder/main.go

```diff
@@ -132,34 +136,87 @@ func runMain() error {
 	}
 }
 
+type HeaderInfo struct {
+	resX      int
+	resY      int
+	fps       int
+	framesize int
+	model     string
+	brand     string
+}
+
+func (h *HeaderInfo) ResX() int {
+	return h.resX
+}
+func (h *HeaderInfo) ResY() int {
+	return h.resY
+}
+func (h *HeaderInfo) FPS() int {
+	return h.fps
+}
+
+func NewHeader(headerInfo map[string]interface{}) *HeaderInfo {
+	return &HeaderInfo{
+		resX:      headerInfo[headers.XResolution].(int),
+		resY:      headerInfo[headers.YResolution].(int),
+		fps:       headerInfo[headers.FPS].(int),
+		framesize: headerInfo[headers.FrameSize].(int),
+		model:     headerInfo[headers.Model].(string),
+		brand:     headerInfo[headers.Model].(string),
+	}
+}
+
+func readHeader(conn net.Conn) (*HeaderInfo, error) {
```

Comment:
The bufio.Reader should be created first, then passed into readHeader and then used for all future reads.

Only creating and using the Reader here means that data beyond the headers can end up being caching into the Reader but will never seen by the frame reading code.